src/App.tsx

```diff
@@ -15,48 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+interface SortParams {
+  sortType: string,
+  isReversed: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortType, isReversed }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case (SortType.Alphabet):
+          return good1.localeCompare(good2);
+
+        case (SortType.Length):
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const isResetButtonVisible = sortType || isReversed;
+  const goods = getPreparedGoods(goodsFromServer, { sortType, isReversed });
```

Comment:
Why are you putting the last two arguments into the object?

If arguments order bother you, then it's better to put everything in an object.